x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/get_custom_dashboard.ts

```diff
@@ -39,22 +39,24 @@ export function initGetCustomDashboardRoute(framework: KibanaFramework) {
 
       if (customDashboards.total === 0) {
         return response.ok({
-          body: InfraGetCustomDashboardsResponseBodyRT.encode({
-            assetType: params.assetType,
-            dashboardIdList: [],
-            kuery: undefined,
-          }),
+          body: InfraGetCustomDashboardsResponseBodyRT.encode([
+            {
+              dashboardSavedObjectId: '',
+              dashboardFilterAssetIdEnabled: false,
+              assetType: params.assetType,
+            },
+          ]),
```

Comment:
Could we return an empty array instead? I guess the client side won't need this dummy object, and if it does, it could perhaps be created on the client side. Returning an empty array makes this endpoint response clearer when there is no data

```suggestion
          body: [],
```